src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -326,8 +331,36 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
         }
     }, [backend, sendAlert]);
 
+    const statusRef = useRef(status);
+
+    const KILL_TIMEOUT = 2500;
+    useEffect(() => {
+        statusRef.current = status;
+        if (status === ExecutionStatus.KILLING) {
+            setTimeout(() => {
+                if (statusRef.current === ExecutionStatus.KILLING) {
+                    log.info('Executor did not actually kill the backend. Forcing restart.');
+
+                    ipcRenderer
+                        .invoke('restart-backend')
```

Comment:
Could you please use the `restart` method from `BackendContext`? That method would probably be a better place to put the `backend.abort();` as well.

Reply:
Whoops, accidentally reverted that when i copied the old code back